test/test-jsdomparser.js

```diff
@@ -6,7 +6,7 @@ var readability = require("../index.js");
 var JSDOMParser = readability.JSDOMParser;
 
 var BASETESTCASE = '<html><body><p>Some text and <a class="someclass" href="#">a link</a></p>' +
-                   '<div id="foo">With a <script>With < fancy " characters in it because' +
+                   '<div id="foo">With a <script>With &lt; fancy " characters in it because' +
```

Comment:
```
I think replacing the < and > that start/end CDATA sections (like my comment below) is definitely wrong.
```
Agreed, let's do it. I'll not replace `<` and `>` that start/end CDATA sections.

But I cannot understand `I think that if it works without the replacement we should leave the files alone.` Do you mean that we can replace `<` and `>` in an embedded script if there is no `CDATA` in it? For example, we can replace the below `<` or `>` with `&lt;` or `&gt;`.
```js
<script>
  // There is no CDATA here.
  for (var i = 0; i < 10; i++) {
    console.log(i);
  }
</script>
```

After doing replacement, it will looks like
```js
<script>
  // There is no CDATA here.
  for (var i = 0; i &lt; 10; i++) {
    console.log(i);
  }
</script>
```

Is that OK to you?